go-controller/pkg/cluster/node.go

```diff
@@ -131,25 +133,68 @@ func (cluster *OvnClusterController) updateOvnNode(masterIP string,
 	return nil
 }
 
-// watchNamespaceUpdate starts watching namespace resources and calls back
-// the update handler logic if there is any namspace update event
-func (cluster *OvnClusterController) watchNamespaceUpdate(node *kapi.Node,
-	subnet string) error {
-	_, err := cluster.watchFactory.AddNamespaceHandler(
-		cache.ResourceEventHandlerFuncs{
-			UpdateFunc: func(old, newer interface{}) {
-				oldNs := old.(*kapi.Namespace)
-				oldMasterIP := oldNs.Annotations[MasterOverlayIP]
-				newNs := newer.(*kapi.Namespace)
-				newMasterIP := newNs.Annotations[MasterOverlayIP]
-				if newMasterIP != oldMasterIP {
-					err := cluster.updateOvnNode(newMasterIP, node, subnet)
-					if err != nil {
-						logrus.Errorf("Failed to update OVN node with new "+
-							"masterIP %s: %v", newMasterIP, err)
-					}
+func validateOVNConfigEndpoint(ep *kapi.Endpoints) bool {
+	if len(ep.Subsets) != 1 {
+		return false
+	}
+	if len(ep.Subsets[0].Ports) != 2 {
+		return false
+	}
+	return true
+
+}
+
+func updateOVNConfig(ep *kapi.Endpoints) {
+	if validateOVNConfigEndpoint(ep) {
+		var southboundDBPort string
+		var northboundDBPort string
+		for _, ovnDB := range ep.Subsets[0].Ports {
+			if ovnDB.Name == "southbound-db" {
+				southboundDBPort = strconv.Itoa(int(ovnDB.Port))
+			}
+			if ovnDB.Name == "northbound-db" {
+				northboundDBPort = strconv.Itoa(int(ovnDB.Port))
+			}
+		}
+		err := config.UpdateOVNNodeAuth(ep.Subsets[0].Addresses[0].IP, &southboundDBPort, &northboundDBPort)
+		if err != nil {
+			logrus.Errorf(err.Error())
+			return
+		}
+		for _, auth := range []config.OvnAuthConfig{config.OvnNorth, config.OvnSouth} {
+			if err := auth.SetDBAuth(); err != nil {
+				logrus.Errorf(err.Error())
+				return
+			}
+			logrus.Infof("OVN databases reconfigured, masterIP %s, northbound-db %s, southbound-db %s", ep.Subsets[0].Addresses[0].IP, northboundDBPort, southboundDBPort)
+		}
+
+	} else {
+		logrus.Errorf("endpoint %s is not in the right format to configure OVN", ep.Name)
+	}
+
+}
+
+//watchEndpoints starts the watching of Endpoint resource and calls back to the appropriate handler logic
+func (cluster *OvnClusterController) watchEndpoints() error {
+	_, err := cluster.watchFactory.AddEndpointsHandler(cache.ResourceEventHandlerFuncs{
```

Comment:
Would be a bit cleaner to add a `AddFilteredEndpointsHandler` function pkg/factory/factory.go that takes a namespace and passes it to wf.addHandler(), and then just call that from watchEndpoints() with `config.Kubernetes.OVNConfigNamespace`